src/cli/self_update.rs

```diff
@@ -1126,15 +1125,41 @@ pub(crate) fn self_update_permitted(explicit: bool) -> Result<SelfUpdatePermissi
     Ok(SelfUpdatePermission::Permit)
 }
 
-/// Performs all of a self-update: check policy, download, apply and exit.
-pub(crate) async fn self_update(dl_cfg: &DownloadCfg<'_>) -> Result<utils::ExitCode> {
+/// Optionally performs a self-update: check policy, download, apply and exit.
+///
+/// Whether the self-update is executed is based on both compile-time and runtime
+/// configurations, where the priority is as follows:
+/// no-self-update feature > self update mode > no-self-update args.
+///
+/// i.e. update only if rustup does **not** have the no-self-update feature,
+/// and self update mode is configured to **enable**
+/// and has **no** no-self-update parameter.
+pub(crate) async fn self_update(
+    dl_cfg: &DownloadCfg<'_>,
+    mode: SelfUpdateMode,
+    no_self_update: bool,
```

Comment:
I'm not sure about this being a double negative. Mistakes have been made in other code bases due to confusion over such things, especially in combination with boolean operators. E.g. `!no_self_update` should be read as "not not self-update" which isn't ideal.

Reply:
I'm okay with reworking on that.

PS: Interestingly, the first version of this PR explicitly works to address some double negatives in a PR and got dropped 🤔